esp/src/src-react/components/forms/Fields.tsx

```diff
@@ -100,35 +192,61 @@ type Field = StringField | CheckboxField | DateTimeField | LinkField |
 
 export type Fields = { [id: string]: Field };
 
-const TargetClusterTextField: React.FunctionComponent<IDropdownProps> = (props) => {
+export interface TargetClusterTextFieldProps extends DropdownProps {
+    key: string;
+    label?: string;
+    selectedKey?: string;
+    className?: string;
+    onChange?: (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption, index?: number) => void;
+    placeholder?: string;
+}
+
+export const TargetClusterTextField: React.FunctionComponent<TargetClusterTextFieldProps> = ({
+    key,
+    label,
+    selectedKey,
+    className,
+    onChange,
+    placeholder
+}) => {
 
     const [targetClusters, setTargetClusters] = React.useState<IDropdownOption[]>([]);
 
     React.useEffect(() => {
         const topology = new Topology({ baseUrl: "" });
         topology.fetchLogicalClusters().then((response: TpLogicalClusterQuery.TpLogicalCluster[]) => {
-            setTargetClusters(
-                [
-                    { Name: "", Type: "", LanguageVersion: "", Process: "", Queue: "" },
-                    ...response
-                ]
-                    .map(n => {
-                        return {
-                            key: n.Name,
-                            text: n.Name + (n.Name !== n.Type ? ` (${n.Type})` : "")
-                        };
-                    })
+            setTargetClusters(response
+                .map((n, i) => {
+                    return {
+                        key: n.Name || "unknown",
+                        text: n.Name + (n.Name !== n.Type ? ` (${n.Type})` : ""),
+                    };
+                })
             );
         });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    return <Dropdown
-        {...props}
-        options={targetClusters}
-    />;
+    return <Dropdown key={key} label={label} selectedKey={selectedKey} optional className={className} onChange={onChange} placeholder={placeholder} options={targetClusters} />;
 };
 
-const TargetGroupTextField: React.FunctionComponent<IDropdownProps> = (props) => {
+export interface TargetGroupTextFieldProps {
+    key: string;
+    label?: string;
+    selectedKey?: string;
+    className?: string;
+    onChange?: (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption, index?: number) => void;
+    placeholder?: string;
+}
+
+const TargetGroupTextField: React.FunctionComponent<TargetGroupTextFieldProps> = ({
```

Comment:
Should this have been exported like the others?

Reply:
Probably...

Reply:
Added missing export